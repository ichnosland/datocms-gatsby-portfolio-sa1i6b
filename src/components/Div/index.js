/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
