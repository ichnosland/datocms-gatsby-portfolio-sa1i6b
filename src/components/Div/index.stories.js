import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import Div from './index';


const nota = 'La props nobreak assegna \'page-break-inside: avoid;\' \n quella noPrint \'display:none;\' in stampa';

storiesOf('Div', module)
  .addDecorator(withKnobs)
  .addWithJSX('Div', withNotes(nota)(() => (
    <Div
      display={text('display', 'block')}
      position={text('position', 'relative')}
      width={text('width', 'auto')}
      maxWidth={text('maxWidth', '100%')}
      height={text('height', '')}
      color={color('color', 'inherit')}
      align={text('align (tex-align)', '')}
      float={text('float', 'none')}
      margin={text('margin', '0')}
      padding={text('padding', '0')}
      radius={text('radius', '')}
      bgColor={color('bgColor', 'transparent')}
      border={text('border', '')}
      borderwidth={text('borderwidth', '')}
      borderstyle={text('borderstyle', '')}
      bordercolor={color('bordercolor', '')}
      overflow={text('overflow', 'visible')}
      userSelect={text('userSelect', 'auto')}
      pointerEvents={text('pointerEvents', 'auto')}
      zindex={text('zindex', 'auto')}
      nobreak={boolean('nobreak', false)}
      noPrint={boolean('noPrint', true)}
    >
      <p>Ciao, sono all&apos;interno di un Div. Usa le props per personalizzarlo :)</p>
    </Div>
  )));
