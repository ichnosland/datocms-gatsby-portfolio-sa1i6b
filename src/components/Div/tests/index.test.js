import React from 'react';
import { shallow } from 'enzyme';

import Div from '../index';

describe('<Div />', () => {
  it('should render a <div> tag', () => {
    const renderedComponent = shallow(<Div />);
    expect(renderedComponent.type()).toEqual('div');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Div id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should render its css w/ props display', () => {
    const renderedComponent = shallow(<Div display="inline-block" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props position', () => {
    const renderedComponent = shallow(<Div position="absolute" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props width', () => {
    const renderedComponent = shallow(<Div width="300px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props maxWidth', () => {
    const renderedComponent = shallow(<Div maxWidth="800px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props height', () => {
    const renderedComponent = shallow(<Div height="150px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props color', () => {
    const renderedComponent = shallow(<Div color="ffcc00" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props align', () => {
    const renderedComponent = shallow(<Div align="center" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props float', () => {
    const renderedComponent = shallow(<Div float="left" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props margin', () => {
    const renderedComponent = shallow(<Div margin="0 auto" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props padding', () => {
    const renderedComponent = shallow(<Div padding="20px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props border', () => {
    const renderedComponent = shallow(<Div border="2px solid red" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props borderstyle', () => {
    const renderedComponent = shallow(<Div borderstyle="dotted" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bordercolor', () => {
    const renderedComponent = shallow(<Div bordercolor="black" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props borderwidth', () => {
    const renderedComponent = shallow(<Div borderwidth="0 0 1em 0" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgColor', () => {
    const renderedComponent = shallow(<Div bgColor="yellow" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props radius', () => {
    const renderedComponent = shallow(<Div radius="60px" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props overflow', () => {
    const renderedComponent = shallow(<Div overflow="hidden" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props userSelect', () => {
    const renderedComponent = shallow(<Div userSelect="none" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props pointerEvents', () => {
    const renderedComponent = shallow(<Div pointerEvents="none" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props zindex', () => {
    const renderedComponent = shallow(<Div zindex="30" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props noPrint', () => {
    const renderedComponent = shallow(<Div noPrint />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props nobreak', () => {
    const renderedComponent = shallow(<Div nobreak />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
