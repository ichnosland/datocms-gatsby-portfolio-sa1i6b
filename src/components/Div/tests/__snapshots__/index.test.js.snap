// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Div /> should render its css w/ props align 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  text-align: center;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props bgColor 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: yellow;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props border 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: 2px solid red;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props bordercolor 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  border-color: black;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props borderstyle 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  border-style: dotted;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props borderwidth 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  border-width: 0 0 1em 0;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props color 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: ffcc00;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
  color="ffcc00"
/>
`;

exports[`<Div /> should render its css w/ props display 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
  display="inline-block"
/>
`;

exports[`<Div /> should render its css w/ props float 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: left;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props height 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  height: 150px;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
  height="150px"
/>
`;

exports[`<Div /> should render its css w/ props margin 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0 auto;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props maxWidth 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 800px;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props noPrint 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

@media print {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props nobreak 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
  page-break-inside: avoid;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props overflow 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: hidden;
}

<div
  className="c0"
  overflow="hidden"
/>
`;

exports[`<Div /> should render its css w/ props padding 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 20px;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props pointerEvents 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
  pointer-events: none;
}

<div
  className="c0"
  pointerEvents="none"
/>
`;

exports[`<Div /> should render its css w/ props position 1`] = `
.c0 {
  display: block;
  position: absolute;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props radius 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  border-radius: 60px;
  overflow: visible;
}

<div
  className="c0"
  radius="60px"
/>
`;

exports[`<Div /> should render its css w/ props userSelect 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
/>
`;

exports[`<Div /> should render its css w/ props width 1`] = `
.c0 {
  display: block;
  position: relative;
  width: 300px;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
}

<div
  className="c0"
  width="300px"
/>
`;

exports[`<Div /> should render its css w/ props zindex 1`] = `
.c0 {
  display: block;
  position: relative;
  max-width: 100%;
  color: inherit;
  float: none;
  margin: 0;
  padding: 0;
  border: none;
  background-color: transparent;
  overflow: visible;
  z-index: 30;
}

<div
  className="c0"
/>
`;
